src/shared_utilities/FileReader.jl

```diff
@@ -0,0 +1,413 @@
+            (all_dates[Int(date_idx + 1)] - all_dates[Int(date_idx)]).value
+        # Read in data fields at both dates
+        pd_file_state.data_fields[1] .= Regridder.swap_space!(
+            Regridder.read_from_hdf5(
+                regrid_dirpath,
+                outfile_root,
+                all_dates[Int(date_idx)],
+                varname,
+                comms_ctx,
+            ),
+            space,
+        )
+        pd_file_state.data_fields[2] .= Regridder.swap_space!(
+            Regridder.read_from_hdf5(
+                regrid_dirpath,
+                outfile_root,
+                all_dates[Int(date_idx + 1)],
+                varname,
+                comms_ctx,
+            ),
+            space,
+        )
+        # Case 4: Nearest date index not being used for initial date field
+        #  Throw warning and update date index
+    elseif Dates.days(date - all_dates[Int(date_idx + 1)]) > 2
+        nearest_idx = argmin(
+            abs.(
+                parse(FT, datetime_to_strdate(date)) .-
+                parse.(FT, datetime_to_strdate.(all_dates[:]))
+            ),
+        )
+        pd_file_state.date_idx[1] = date_idx = date_idx0 = nearest_idx
+        @warn "init data does not correspond to start date. Initializing with `PrescribedData.date_idx[1] = date_idx = date_idx0 = $nearest_idx` for this start date"
+    else
+        throw(ErrorException("Check boundary file specification"))
+    end
+end
+
+"""
+    next_date_in_file(prescribed_data)
+
+Returns the next date stored in the file `prescribed_data` struct after the
+current date index given by `date_idx`.
+Note: this function does not update `date_idx`, so repeated calls will
+return the same value unless `date_idx` is modified elsewhere in between.
+
+# Arguments
+- `prescribed_data`: [PrescribedData] containing the date information.
+
+# Returns
+- Dates.DateTime
+"""
+next_date_in_file(prescribed_data::PrescribedData) =
+    prescribed_data.file_info.all_dates[prescribed_data.file_state.date_idx[1] + Int(
+        1,
+    )]
+
+"""
+    interpolate_data(prescribed_data::PrescribedData, date, space)
+
+Interpolates linearly between two `Fields` in the `prescribed_data` struct,
+or returns the first Field if interpolation is switched off.
+
+# Arguments
+- `prescribed_data`: [PrescribedData] contains fields to be interpolated.
+- `date`: [Dates.DateTime] start date for data.
+- `space`: [Spaces.AbstractSpace] the space of our simulation.
+
+# Returns
+- Fields.field
+"""
+function interpolate_data(prescribed_data::PrescribedData, date, space)
+    FT = Spaces.undertype(space)
+    # Interpolate if the time period between dates is nonzero
+    if prescribed_data.file_state.segment_length[1] > FT(0)
+        (; segment_length, date_idx, data_fields) = prescribed_data.file_state
+        (; all_dates) = prescribed_data.file_info
+
+        return interpol.(
+            Regridder.swap_space!(data_fields[1], space),
```

Comment:
I think we do not need this here now, though? (swap_space!)

Reply:
Good catch